### PR TITLE
Emphasize S3-S4 difference in recommendation for how to fix class_equals_linter()

### PR DESCRIPTION
**Pull request by MichaelChirico**
Per experience here:

https://github.com/Rdatatable/data.table/pull/6603#discussion_r1854842225

I think it's good to more strongly recommend `is()` only for S4 classes.

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Looks like there's on ongoing outage for `r-lib/actions` on `macos: latest`:

https://github.com/r-lib/actions/issues/948

**Comment by AshesITR** [comment]
How do you feel about a more subtle nudge by mentioning `is(x, "s4-class")`?

**Comment by MichaelChirico** [comment]
It crossed my mind, but OTOH I don't see a big benefit to being overly parsimonious with the lint description, and the comment I received was that they had totally glanced over `inherits()`. `"s4-class"` _could_ be enough, though. If you feel the current description is definitely too wordy I can make the change.

**Review by AshesITR — APPROVED** [review]
It's fine both ways for me.
